models/location_helpers.go

```diff
@@ -111,18 +111,93 @@ func FindBackupLocations(q *reform.Querier) ([]*BackupLocation, error) {
 	return locations, nil
 }
 
+// FindBackupLocationByID finds Location by ID.
+func FindBackupLocationByID(q *reform.Querier, id string) (*BackupLocation, error) {
+	if id == "" {
+		return nil, status.Error(codes.InvalidArgument, "Empty Location ID.")
+	}
+
+	location := &BackupLocation{ID: id}
+	switch err := q.Reload(location); err {
+	case nil:
+		return location, nil
+	case reform.ErrNoRows:
+		return nil, status.Errorf(codes.NotFound, "Backup location with ID %q not found.", id)
+	default:
+		return nil, errors.WithStack(err)
+	}
+}
+
+// BackupLocationsConfigs groups all backup locations configs.
+type BackupLocationsConfigs struct {
+	PMMClientConfig *PMMClientLocationConfig
+	PMMServerConfig *PMMServerLocationConfig
+	S3Config        *S3LocationConfig
+}
+
+// Validate checks if there is exactly one config with required fields.
+func (c BackupLocationsConfigs) Validate() error {
+	configCount := 0
+	if c.S3Config != nil {
+		configCount++
+		if err := checkS3Config(c.S3Config); err != nil {
+			return err
+		}
+	}
+	if c.PMMServerConfig != nil {
+		configCount++
+		if err := checkPMMServerLocationConfig(c.PMMServerConfig); err != nil {
+			return err
+		}
+	}
+	if c.PMMClientConfig != nil {
+		configCount++
+		if err := checkPMMClientLocationConfig(c.PMMClientConfig); err != nil {
+			return err
+		}
+	}
+
+	if configCount > 1 {
+		return status.Error(codes.InvalidArgument, "Only one config is allowed.")
+	}
+
+	return nil
+}
+
+// FillLocation fills provided location based on config.
```

Comment:
I'd clarify this comment: sets the pointed location's data according to the backup locations config.